test/functional/terminal/buffer_spec.lua

```diff
@@ -224,6 +225,23 @@ describe(':terminal buffer', function()
       neq('terminal', eval('&buftype'))
     end)
   end)
+
+  it('it works with set rightleft #11438', function()
+    local columns = eval('&columns')
+    feed(string.rep('a', columns))
+    command('set rightleft')
+    wait()
```

Comment:
did you try without `wait()`? Usually not needed before `screen:expect` .

Reply:
Oops! It is not necessary because it is a remnant during trial and error.